Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/DirectILL_common.py

```diff
@@ -188,8 +188,12 @@ def get_grouping_pattern(ws, vertical_step, horizontal_step=1):
         while pixel_id < n_pixels - (group_by_x - 1) * n_pixels_per_tube:
             pattern = []
             for tube_shift in range(0, group_by_x):
+                if n_pixels < pixel_id + tube_shift * n_pixels_per_tube + group_by_y:
```

Comment:
This 'if' condtition could be replaced by `pattern_max = min(n_pixels, pixel_id + tube_shift * n_pixels_per_tube + group_by_y` . IMO, that would render the loop at little more digest.

Reply:
Good point, thanks.